src/modules/variable/set.rs

```diff
@@ -65,23 +75,19 @@ impl TranslateModule for VariableSet {
     fn translate(&self, meta: &mut TranslateMetadata) -> String {
         let name = self.name.clone();
         let index = self.index.as_ref()
-            .map(|index| format!("[{}]", self.is_ref
-                .then(|| index.translate_eval(meta, true))
-                .unwrap_or_else(|| index.translate(meta))))
+            .map(|index| self.translate_eval_if_ref(index, meta))
+            .map(|index| format!("[{index}]"))
             .unwrap_or_default();
-        let mut expr = self.is_ref
-            .then(|| self.value.translate_eval(meta, true))
-            .unwrap_or_else(|| self.value.translate(meta));
-        if let Type::Array(_) = self.value.get_type() {
-            expr = format!("({})", expr);
+        let mut expr = self.translate_eval_if_ref(self.expr.as_ref(), meta);
+        if let Type::Array(_) = self.expr.get_type() {
+            expr = format!("({expr})");
         }
-        if self.is_ref {
+        if let Some(id) = self.global_id {
```

Comment:
We previously tested `self.global_id` _after_ `self.is_ref`, but I switched the order to match the shorthand operators.  I think this is safe, because `global_id` and `is_ref` cannot both be enabled at the same time.